CMakeLists.txt

```diff
@@ -54,24 +53,30 @@ add_library(graphqlservice
   GraphQLResponse.cpp
   GraphQLService.cpp
   Introspection.cpp
-  ${CMAKE_BINARY_DIR}/graphqlservice/IntrospectionSchema.cpp)
+  IntrospectionSchema.cpp)
 target_link_libraries(graphqlservice PRIVATE taocpp::pegtl)
 target_link_libraries(graphqlservice PUBLIC Threads::Threads)
-target_include_directories(graphqlservice
-  PUBLIC
-    $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/include>
-    $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}>
-    $<INSTALL_INTERFACE:include>
-)
+# Make system includes (e.g. <graphqlservice/public.h>) work relative to the build/install generator expressions
+target_include_directories(graphqlservice SYSTEM PUBLIC
```

Comment:
I don't think `SYSTEM` should be used.
It will shutdown compiler warnings because the compiler will think it is a system header.
It is perfectly fine to use angled brackets for non-system headers, the only meaning of double-quoted includes is that a lookup relative to the current source file is first performed.

I use `SYSTEM` only when I vendor a third party library which has a lot of noisy warnings.